src/pyinit.jl

```diff
@@ -190,6 +190,24 @@ function __init__()
         Py_SetPythonHome(libpy_handle, pyversion, pyhome)
         Py_SetProgramName(libpy_handle, pyversion, current_python())
         ccall((@pysym :Py_InitializeEx), Cvoid, (Cint,), 0)
+
+        # _pydecref_queue used to pydecref only on the main thread:
+        @async begin
+            while !_finalized[]
+                if !isempty(_pydecref_queue)
+                    lock(_pydecref_queue_lock)
+                    o = try
+                        pop!(_pydecref_queue, o)
+                    finally
+                        unlock(_pydecref_queue_lock)
+                    end
+                    pydecref(o)
+                else
+                    yield()
```

Comment:
I'm not sure I like the thought of putting the CPU in a spinloop calling `yield()` over and over any time PyCall is loaded…

Reply:
(The basic principle of the FFTW.jl solution is that you can flush the queue of objects to finalize the next time you have a finalization that succeeds in grabbing the lock.)